packages/edit-site/src/components/dataviews/item-actions.js

```diff
@@ -6,12 +6,67 @@ import {
 	MenuGroup,
 	MenuItem,
 	Button,
+	Modal,
 	__experimentalHStack as HStack,
 } from '@wordpress/components';
 import { __ } from '@wordpress/i18n';
-import { useMemo } from '@wordpress/element';
+import { useMemo, useState } from '@wordpress/element';
 import { moreVertical } from '@wordpress/icons';
 
+function PrimaryActionTrigger( { action, onClick } ) {
+	return (
+		<Button
+			label={ action.label }
+			icon={ action.icon }
+			isDestructive={ action.isDestructive }
+			size="compact"
+			onClick={ onClick }
+		/>
+	);
+}
+
+function SecondaryActionTrigger( { action, onClick } ) {
+	return (
+		<MenuItem onClick={ onClick } isDestructive={ action.isDestructive }>
+			{ action.label }
+		</MenuItem>
+	);
+}
+
+function MaybeWithModal( { action, item } ) {
+	const [ isModalOpen, setIsModalOpen ] = useState( false );
+	const actionTriggerProps = {
+		action,
+		onClick: action.modalProps ? setIsModalOpen : action.perform,
+	};
+	const ActionTrigger = action.isPrimary
+		? PrimaryActionTrigger
+		: SecondaryActionTrigger;
+	if ( ! action.modalProps ) {
+		return <ActionTrigger { ...actionTriggerProps } />;
+	}
+	const { ModalContent, ...modalProps } = action.modalProps;
+	return (
+		<>
+			<ActionTrigger { ...actionTriggerProps } />
+			{ isModalOpen && (
+				<Modal
+					{ ...modalProps }
+					onRequestClose={ () => {
+						setIsModalOpen( false );
+					} }
+					__experimentalHideHeader={ ! modalProps.title }
+				>
+					<ModalContent
+						item={ item }
+						setIsModalOpen={ setIsModalOpen }
```

Comment:
Instead of passing `setIsModalOpen` can we pass `closeModal` without parameters? So the consumer only has to do `closeModal()` instead of `setIsModalOpen( false )`.